backend/apps/notificationpref/services.py

```diff
@@ -0,0 +1,50 @@
+from apps.notificationpref.models import UserNotificationPreference
+from apps.users.models import BaseUser
+from apps.communities.models import Community
+
+def notification_preference_create(
+    *, user: BaseUser, subscribed_communities: list = [], event_updates: bool = True,
+    post_notifications: bool = True, announcements: bool = True, email_notifications: bool = True,
+    in_app_notifications: bool = True
+) -> UserNotificationPreference:
+    
+    preference, created = UserNotificationPreference.objects.get_or_create(
+        user=user,
+        defaults={
+            "event_updates": event_updates,
+            "post_notifications": post_notifications,
+            "announcements": announcements,
+            "email_notifications": email_notifications,
+            "in_app_notifications": in_app_notifications,
+        }
+    )
+
+    if created or subscribed_communities:
+        preference.subscribed_communities.set(subscribed_communities)
+        preference.save()
+
+    return preference
+
+def notification_preference_update(*, user: BaseUser, data: dict) -> UserNotificationPreference:
```

Comment:
There is a function called model_update, which handles updating models (and will update all models the same, therefore making it a bit safer/easier to keep track of bugs). 

See user/services.py and update this to follow similarly